src/compojure/api/meta.clj

```diff
@@ -59,7 +61,7 @@
         (if-let [matcher (:response (mw/get-coercion-matcher-provider request))]
           (let [body (schema/coerce schema (:body response) matcher)]
             (if (schema/error? body)
-              (internal-server-error {:errors (:error body)})
+              (throw+ (assoc body :type ::ex/response-validation))
```

Comment:
src-coerce! in `meta.clj` should also use plain `coerce` and throw `::ex/request-validation`.